app/models/postcode_local_restriction_search.rb

```diff
@@ -0,0 +1,53 @@
+class PostcodeLocalRestrictionSearch
+  UK_POSTCODE_PATTERN = %r{
+    \A
+    ([Gg][Ii][Rr] 0[Aa]{2}) # Unusual postcodes
+    |
+    (
+      # Outward code
+      (([A-Za-z][0-9]{1,2})|(([A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2})|(([A-Za-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y][0-9][A-Za-z]?))))
+      \s?
+      [0-9][A-Za-z]{2} # Inward code
+    )
+    \Z
+  }x.freeze
+
+  attr_reader :query, :normalised_postcode
+  delegate :no_information?, to: :location_lookup
+
+  def initialize(query)
+    @query = query
+    # Use the uk_postcode gem to potentially transpose O/0 and I/1.
+    @normalised_postcode = UKPostcode.parse(
+      query.gsub(/[^a-z0-9 ]/i, "").strip,
```

Comment:
I've replied in the previous comment regarding the bomb concern.

I've moved the normalising into a postcode.

Using `sanitized` seemed a dubious choice to me which is why I went with normalised. It seems unconventional for us to use because it has the US spelling and using the term [sanity is divisive](https://developers.google.com/style/inclusive-documentation#ableist-language).

It's not something I feel too strongly about so can change it to sanitized if you're convinced the consistency makes it easier to understand.